File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        if not word1:
            return word2
        if not word2:
            return word1
        l = 0
        r = 0
        s = ""
        while l < len(word1) and r < len(word2):
            s +=(word1[l])
            s+=(word2[r])
            l+=1
            r+=1
             # Append remaining characters from word1
        while l < len(word1):
            s += word1[l]
            l += 1
        
        # Append remaining characters from word2
        while r < len(word2):
            s += word2[r]
            r += 1
        
        return s

File: 1894-merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let s = ""
    let l = 0
     let r = 0
     while(l < word1.length && r < word2.length){
        s += word1[l]
        s+= word2[r]
        l++
        r++
     }
     while(l < word1.length){
        s+= word1[l]
        l++
     }
     while(r< word2.length){
        s+= word2[r]
        r++
     }
     return s;
};